homestar-runtime/src/event_handler.rs

```diff
@@ -110,17 +123,23 @@ where
         ws_msg_sender: ws::Notifier,
     ) -> Self {
         let (sender, receiver) = Self::setup_channel(settings);
+        let sender = Arc::new(sender);
         Self {
             receipt_quorum: settings.network.receipt_quorum,
             workflow_quorum: settings.network.workflow_quorum,
             p2p_provider_timeout: settings.network.p2p_provider_timeout,
             db,
             swarm,
-            sender: Arc::new(sender),
+            cache: Arc::new(setup_cache(sender.clone())),
+            sender: sender.clone(),
```

Comment:
did we need to arc outside and clone in here?

Reply:
Was able to simplify it to this:

https://github.com/ipvm-wg/homestar/blob/402a4121587bda974c27ec8c93dc86538aa61a06/homestar-runtime/src/event_handler.rs#L133-L134